Framework/Algorithms/src/RemoveBackground.cpp

```diff
@@ -123,11 +127,11 @@ void RemoveBackground::exec() {
   for (int hist = 0; hist < histnumber; ++hist) {
     PARALLEL_START_INTERUPT_REGION
     // get references to output Workspace X-arrays.
-    MantidVec &XValues = outputWS->dataX(hist);
+    auto &XValues = outputWS->mutableX(hist);
     // get references to output workspace data and error. If it is new
     // workspace, data will be copied there, if old, modified in-place
-    MantidVec &YValues = outputWS->dataY(hist);
-    MantidVec &YErrors = outputWS->dataE(hist);
+    auto &YValues = outputWS->mutableY(hist);
+    auto &YErrors = outputWS->mutableE(hist);
```

Comment:
(having had a look at `BackgroundHelper` below it may not be worth it, because there is a mess there anyway that is too much effort to do in a nicer way as part of these changes.)